### PR TITLE
Restructure code and highlight states at State level

### LINKED ISSUE #87: On state level, hovering other states should highlight them
The hover should show the name of the state. When clicking on it, it should switch the view to the selected state.

Bonus: hover highlight also possible when choosing state from combobox

### LINKED ISSUE #100: Clicking on neighboring state in state level does not select it

### PR DESCRIPTION
**Pull request by Bachibouzouk**
Fix #100 
Fix #87

### CODE REVIEW

File: app/static/js/layers.js [hunk 1]
```diff
@@ -208,18 +164,68 @@ function update_lgas_pbf(){
...
+  var avail = {
+    gridTracking: "<b>✕</b>",
+    remoteMapping: "<b>✕</b>",
+    Surveying: "<b>✕</b>",
+  }
+  if (e.target.feature.properties.availability >= 4) {avail.gridTracking = "<b>✓</b>";}
+  if (e.target.feature.properties.availability % 4 >= 2) {avail.remoteMapping = "<b>✓</b>";}
+  if (e.target.feature.properties.availability % 2 === 1) {avail.Surveying = "<b>✓</b>";}
+  highlightLayer = e.target;
+  if (highlightLayer.feature.properties.name != selectedState){
```
Comment (Bachibouzouk) [inline app/static/js/layers.js:170]: Here is the selective highlight which is desactivated if the state is selected

File: app/static/js/sidebar.js [hunk 2]
```diff
@@ -103,96 +114,113 @@ noUiSlider.create(ogPopulationSlider, {
 });
 
 
-function adapt_sidebar_to_national_level() {
-  var hidelist = document.getElementsByClassName("n_hide");
-  var showlist = document.getElementsByClassName("n_show");
+function adapt_sidebar_to_selection_level(selectionLevel) {
+
```
Comment (Bachibouzouk) [inline app/static/js/sidebar.js:118]: I groupped things together logically, I hope the order was not important

**Comment by christian-rli** [comment]
Thank you @Bachibouzouk . Layering seems to work nicely. Only the grid of the previous state does not seem to get removed upon click on a neighbouring state. I'll have a look into that. Should be just a one-line fix.

**Comment by Bachibouzouk** [comment]
> I'll have a look into that. Should be just a one-line fix.

Cool :)

**Comment by christian-rli** [comment]
... looks like it's not quite as easy as expected. Clicking on a state is now an event that triggers the selected state to be changed, is that correct? So where/when would I go about removing the old grid. I wanted to remove the old one upon click, but it's too late then, it appears.

**Comment by christian-rli** [comment]
ready to merge @Bachibouzouk ?

**Pull request merged**